tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -863,10 +863,26 @@ def cleanup_data(self):
         self.device.file_manager.remove('/data/local/storage/permanent')
         self.device.file_manager.remove('/data/local/storage/persistent')
         self.device.file_manager.remove('/data/local/storage/default')
-        self.device.file_manager.remove('/data/local/webapps')
         # remove remembered networks
         self.device.file_manager.remove('/data/misc/wifi/wpa_supplicant.conf')
 
+        if self.device.is_android_build:
+            file_items = self.device.file_manager.list_items('/data/local/webapps/')
+            apps = json.loads(self.device.file_manager.pull_file('/data/local/webapps/webapps.json'))
+            # All default apps have the same installTime, newly added webapps have a different installTime
+            # This difference is used here to selectively delete those webapp directories
+            # Also delete all directories that are not mentioned in the webapps.json file
+            for dir in file_items:
+                delete_file = True
+                for app in apps:
+                    if (dir == str(apps[app]['id']) and apps[app]['installTime'] == apps['system.gaiamobile.org']['installTime']):
+                        print(str(apps[app]['id']))
+                        delete_file = False
+                        break
+                if delete_file and self.device.file_manager.dir_exists('/data/local/webapps/' + dir):
+                    self.device.file_manager.remove('/data/local/webapps/' + dir)
+                    print('deleting dir for: ' + dir + ':: ' + str(apps[app]['installTime']))
```

Comment:
I don't think we need to list the items in `/data/local/webapps` as we should be able to extract that from the `basePath` from the JSON file. I haven't tested it, but the following should work:

```python
apps = json.loads(self.device.file_manager.pull_file('/data/local/webapps/webapps.json'))
system_install_time = apps['system.gaiamobile.org']['installTime']
for app in apps.values():
    if app.get('installTime') > system_install_time:
        # removing any webapps installed since build time
        path = posixpath.join(app.get('basePath'), app.get('id')
        self.logger.debug('Removing %s' % path)
        self.device.file_manager.remove(path)
```